schema.py

```diff
@@ -266,22 +267,34 @@ def is_valid(self, data):
         else:
             return True
 
+    def _raise_error(self, exception, message, *error_list):
+        """Raises exception with custom schema name prepended to exception
+        message and feeds additional arguments into schema error class.
+        """
+        if not self._name and not error_list:
+            raise exception("%s" % (message,))
+        elif not self._name and error_list:
+            raise exception("%s" % (message,), error_list[0])
+        elif self._name and error_list:
+            raise exception("%r %s" % (self._name, message,),
+                            error_list[0])
+
     def validate(self, data):
         Schema = self.__class__
-        s = self._schema
-        e = self._error
         i = self._ignore_extra_keys
-        flavor = _priority(s)
+        def format_error(e): return e.format(data) if e else None
+        flavor = _priority(self._schema)
         if flavor == ITERABLE:
-            data = Schema(type(s), error=e).validate(data)
-            o = Or(*s, error=e, schema=Schema, ignore_extra_keys=i)
+            data = Schema(type(self._schema), error=self._error).validate(data)
+            s = self._schema
```

Comment:
No need for the assignment here.

Reply:
The reason for this assignment is for line 290:
`o = Or(*`__s__`, error=self._error, schema=Schema, ignore_extra_keys=i)`

`s` is used in this assignment expression

Reply:
@SnapperGee why not do `*self._schema` instead of put the assignment on the previous line.